crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -2808,13 +2808,94 @@ impl Analyzer<'_, '_> {
     /// Should be called only if `to` is not expandable.
```

Comment:
Not sure about this, I was expecting to get errors (since im not checking if to is not expandandable) but I didn't get any. Any pointers would be helpful

Reply:
I also tried
```rs
let a = &self.expand(r.span(), Type::StringMapping(to.clone()), Default::default());

        if let Ok(Type::StringMapping(var)) = a {
            match to.kind {
```
and that behaved the same way so I ended up not using it because it seems wasteful. I tried to find an example where not checking whether `to` is expandable would be an issue but I couldnt (for now the function `.assign_to_intrinsic` is only called in one place, perhaps if it is called in more it can become an issue?).

Reply:
Based on the comment here https://github.com/dudykr/stc/blob/af5fda3c18a132ee7959ffb236bfc1879f41951d/crates/stc_ts_type_checker/tests/conformance/types/literal/stringMappingOverPatternLiterals.ts#L74-L79 I think we actually don't want to expand the intrinsic type

Reply:
The comment is not about an error. It means you are not expected to call it if `to` is expandable.

Reply:
And sorry for late response, I missed it.